src/generator.cs

```diff
@@ -1524,8 +1526,23 @@ public TrampolineInfo MakeTrampoline (Type t)
 					string marshal = string.Empty;
 					if (nt == TypeManager.System_Boolean)
 						marshal = "[System.Runtime.InteropServices.MarshalAs (System.Runtime.InteropServices.UnmanagedType.I1)] ";
-					pars.AppendFormat ("{3}{0} {1} {2}", pi.IsOut ? "out" : "ref", FormatType (null, nt), pi.Name.GetSafeParamName (), marshal);
-					invoke.AppendFormat ("{0} {1}", pi.IsOut ? "out" : "ref", pi.Name.GetSafeParamName ());
+					string fnt;
+					string invoke_name;
+					string arg_byref = pi.IsOut ? "out" : "ref";
+					var nullable = TypeManager.GetUnderlyingNullableType (nt);
+					if (nullable != null) {
+						fnt = "IntPtr";
+						invoke_name = "p1";
```

Comment:
This will break if there are more than one parameters like this, I think maybe something like this would be better:

    invoke_name = pi.Name.GetSafeParamName () + "__xamarin_nullified__";

or maybe just:

    invoke_name = "__xamarin_nullified__" + pi.Position;